src/login/LoginOptionsScreen.tsx

```diff
@@ -0,0 +1,215 @@
+import FullScreenHeader from '@atb/components/screen-header/full-header';
+import {StyleSheet} from '@atb/theme';
+import {LoginTexts, useTranslation} from '@atb/translations';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, Linking, ScrollView, View} from 'react-native';
+import {useAuthState} from '@atb/auth';
+import ThemeText from '@atb/components/text';
+import {VippsSignInErrorCode} from '@atb/auth/AuthContext';
+import {RouteProp, useNavigation} from '@react-navigation/native';
+import {StaticColorByType} from '@atb/theme/colors';
+import {
+  getOrCreateVippsUserCustomToken,
+  authorizeUser,
+} from '@atb/api/vipps-login/api';
+import {useAppStateStatus} from '@atb/utils/use-app-state-status';
+import {parseUrl} from 'query-string';
+import storage from '@atb/storage';
+import {LoginInAppStackParams} from '@atb/login/in-app/LoginInAppStack';
+import {AfterLoginParams} from '@atb/login/types';
+import {VippsLoginButton} from '@atb/components/vipps-login-button';
+import MessageBox from '@atb/components/message-box';
+import * as Sections from '@atb/components/sections';
+
+const themeColor: StaticColorByType<'background'> = 'background_accent_0';
+
+export type LoginOptionsRouteParams = {
+  afterLogin: AfterLoginParams;
+};
+
+type LoginOptionsRouteProps = RouteProp<
+  LoginInAppStackParams,
+  'LoginOptionsScreen'
+>;
+
+type LoginOptionsProps = {
+  route: LoginOptionsRouteProps;
+};
+
+export default function LoginOptionsScreen({
+  route: {
+    params: {afterLogin},
+  },
+}: LoginOptionsProps) {
+  const {t} = useTranslation();
+  const styles = useThemeStyles();
+  const {signInWithCustomToken} = useAuthState();
+  const [error, setError] = useState<VippsSignInErrorCode>();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigation = useNavigation();
+  const appStatus = useAppStateStatus();
+  const [authorizationCode, setAuthorizationCode] = useState('');
+
+  const authenticateUserByVipps = async () => {
+    setIsLoading(true);
+    try {
+      await authorizeUser(setIsLoading);
+    } catch (err) {
+      setError('unknown_error');
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(
```

Comment:
this should probably provide a cleanup function

Reply:
I think it already returns the function to unsubscribe from the listener.